src/app/public/src/modules/shared/search-results-provider.ts

```diff
@@ -0,0 +1,143 @@
+import { Injectable } from '@angular/core';
+import { SkyAuthHttp } from '@blackbaud/skyux-builder/runtime';
+import { StacheConfigService } from './config.service';
+import { StacheSearchResult } from './search-result';
+import 'rxjs/add/operator/toPromise';
+
+@Injectable()
+export class StacheSearchResultsProvider {
+
+  private moreResultsUrl: string;
+  private moreResultsLabel: string = 'Show More';
+  private htmlFields: any = {
+    description: true
+  };
+  private isInternal: boolean = true;
+  private siteNames: string[] = [];
+  private endpoints: any = {
+    internal: 'query',
+    public: 'query-public'
+  };
+  private queryUrlBase: string = 'https://stache-search-query.sky.blackbaud.com';
+
+  constructor(
+    private http: SkyAuthHttp,
```

Comment:
Can you do a conditional setting of http?  Try to set the optional `SkyAuthHttp`, then in the constructor, if http does not exist, set it to a new instance of Http from angular?
`@Optional() http = SkyAuthHttp`  
```
if (!http) {
  this.http = new Http();
}
```